### PR TITLE
add ef integration test

### PR DESCRIPTION
**Pull request by SimonCropp**
#skip-changelog

### CODE REVIEW

**Comment by SimonCropp** [comment]
@bruno-garcia @mattjohnsonpint this is ready for a review

**Comment by mattjohnsonpint** [comment]
Re-ran the checks to pass the intermittent error on macOS (#1516), but now it fails intermittently on Windows in `SqlListenerTests.RecordsEf`.  Looks like the order the spans are in is not consistent.

image

**Comment by SimonCropp** [comment]
@mattjohnsonpint that should be fixed now

File: test/Sentry.DiagnosticSource.IntegrationTests/SqlListenerTests.cs [hunk 1]
```diff
@@ -14,11 +14,11 @@ static SqlListenerTests()
             buildTemplate: TestDbBuilder.CreateTable);
     }
 
+#if !NET48
```
Comment (mattjohnsonpint) [inline test/Sentry.DiagnosticSource.IntegrationTests/SqlListenerTests.cs:17]: - nit: `#if !NETFRAMEWORK` would be better, so the framework version is only in the csproj
- q: Why is it that we can't run this test on .NET Framework anyway? I see that if I try, it doesn't capture any spans.  Does the SqlClient not implement `DiagnosticListener` on .NET Framework implementation?  I hunted, but couldn't confirm that.  I thought the diagnostic source stuff was available across the board, perhaps not?

File: test/Sentry.DiagnosticSource.IntegrationTests/Sentry.DiagnosticSource.IntegrationTests.csproj [hunk 2]
```diff
@@ -1,9 +1,26 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFrameworks>net6.0;net5.0;netcoreapp3.1</TargetFrameworks>
+    <TargetFrameworks>net6.0;net5.0;netcoreapp3.1;net48</TargetFrameworks>
```
Comment (mattjohnsonpint) [inline test/Sentry.DiagnosticSource.IntegrationTests/Sentry.DiagnosticSource.IntegrationTests.csproj:4]: Is `net48` the right target? Would an earlier one be more appropriate? Does it matter really? 😉

File: test/Sentry.DiagnosticSource.IntegrationTests/Sentry.DiagnosticSource.IntegrationTests.csproj [hunk 3]
```diff
@@ -1,9 +1,26 @@
...
+    <PackageReference Include="Microsoft.EntityFrameworkCore.SqlServer" Version="3.1.20" />
+  </ItemGroup>
+  <ItemGroup Condition="'$(TargetFramework)' == 'net48'">
+    <ProjectReference Include="../../src/Sentry.DiagnosticSource/Sentry.DiagnosticSource.csproj" />
+    <PackageReference Include="Microsoft.EntityFrameworkCore" Version="3.1.20" />
+    <PackageReference Include="Microsoft.EntityFrameworkCore.SqlServer" Version="3.1.20" />
+  </ItemGroup>
   <ItemGroup>
     <ProjectReference Include="../Sentry.Testing/Sentry.Testing.csproj" />
     <PackageReference Include="LocalDb" Version="13.3.1" />
```
Comment (mattjohnsonpint) [inline test/Sentry.DiagnosticSource.IntegrationTests/Sentry.DiagnosticSource.IntegrationTests.csproj:26]: nit: There are some newer versions of the nuget packages available now.

**Review by mattjohnsonpint — APPROVED** [review]

File: test/Sentry.DiagnosticSource.IntegrationTests/SqlListenerTests.cs (hunk 1 above)
Comment (SimonCropp) [reply test/Sentry.DiagnosticSource.IntegrationTests/SqlListenerTests.cs:17]: > #if !NETFRAMEWORK

yep. it is better

> Why is it that we can't run this test on .NET Framework anyway

https://github.com/dotnet/SqlClient/issues/1529

File: test/Sentry.DiagnosticSource.IntegrationTests/Sentry.DiagnosticSource.IntegrationTests.csproj (hunk 3 above)
Comment (SimonCropp) [reply test/Sentry.DiagnosticSource.IntegrationTests/Sentry.DiagnosticSource.IntegrationTests.csproj]: fixed

File: test/Sentry.DiagnosticSource.IntegrationTests/Sentry.DiagnosticSource.IntegrationTests.csproj (hunk 2 above)
Comment (SimonCropp) [reply test/Sentry.DiagnosticSource.IntegrationTests/Sentry.DiagnosticSource.IntegrationTests.csproj]: TBH i am not sure. i added a 461 just to be sure

**Pull request merged**